src/test/java/com/connexta/ingest/IngestITests.java

```diff
@@ -54,47 +135,99 @@
   @Value("${endpoints.transform.version}")
   private String endpointsTransformVersion;
 
-  @Inject private MockMvc mvc;
-  @Inject private RestTemplate restTemplate;
+  @Value("${s3.bucket}")
+  private String s3Bucket;
+
+  @Inject private AmazonS3 amazonS3;
+
+  @Inject
+  @Named("storeClientRestTemplate")
+  private RestTemplate storeClientRestTemplate;
 
   @Inject
-  @Named("nonBufferingRestTemplate")
-  private RestTemplate nonBufferingRestTemplate;
+  @Named("transformClientRestTemplate")
+  private RestTemplate transformClientRestTemplate;
 
-  private MockRestServiceServer storeServer;
-  private MockRestServiceServer transformServer;
+  private MockRestServiceServer storeMockRestServiceServer;
+  private MockRestServiceServer transformMockRestServiceServer;
+
+  @Inject private WebTestClient webTestClient;
+
+  @TestConfiguration
+  static class Config {
+
+    @Bean
+    public AmazonS3Configuration testAmazonS3Configuration() {
+      return new AmazonS3Configuration(
+          String.format(
+              "http://%s:%d",
+              minioContainer.getContainerIpAddress(), minioContainer.getMappedPort(MINIO_PORT)),
+          "local",
+          MINIO_ADMIN_ACCESS_KEY,
+          MINIO_ADMIN_SECRET_KEY);
+    }
+  }
 
   @BeforeEach
   public void beforeEach() {
-    storeServer = MockRestServiceServer.createServer(nonBufferingRestTemplate);
-    transformServer = MockRestServiceServer.createServer(restTemplate);
+    amazonS3.createBucket(s3Bucket);
+    storeMockRestServiceServer = MockRestServiceServer.createServer(storeClientRestTemplate);
+    transformMockRestServiceServer =
+        MockRestServiceServer.createServer(transformClientRestTemplate);
   }
 
   @AfterEach
   public void afterEach() {
-    storeServer.verify();
-    storeServer.reset();
-    transformServer.verify();
-    transformServer.reset();
+    storeMockRestServiceServer.verify();
+    storeMockRestServiceServer.reset();
+    transformMockRestServiceServer.verify();
+    transformMockRestServiceServer.reset();
+
+    amazonS3.listObjects(s3Bucket).getObjectSummaries().stream()
+        .map(S3ObjectSummary::getKey)
+        .forEach(key -> amazonS3.deleteObject(s3Bucket, key));
+    amazonS3.deleteBucket(s3Bucket);
   }
 
   @Test
   public void testContextLoads() {}
 
+  /* START ingest request tests */
+
   @Test
   public void testSuccessfulIngestRequest() throws Exception {
-    final URI location = new URI("http://localhost:1232/store/1234");
-    storeServer
+    // given
+    final URI location = new URI(endpointUrlStore + "1234");
```

Comment:
:pencil2:  Avoid using string concatenation to build URIs/URLs. Use a URI/URL constructor a builder like UriBuilder or UriComponentsBuilder to create an object. Specialized classes offer better safety and a richer interface than strings.

Reply:
Good suggestion. We can address that in a future PR because there are a few other places that need to be fixed as well.